web/src/components/product/ProductRegistrationPage.test.tsx

```diff
@@ -168,87 +316,163 @@ describe("ProductRegistrationPage", () => {
+            recommended: false,
+            description: "SUSE Linux High Availability Extension provides ...",
+            type: "extension",
+            release: "beta",
           },
         ];
       });
 
-      it("renders registration information with code partially hidden", async () => {
-        const { user } = installerRender(<ProductRegistrationPage />, { withL10n: true });
+      it("renders them", async () => {
+        installerRender(<ProductRegistrationPage />, { withL10n: true });
 
-        // the second "Show" button, the first one belongs to the base product registration code
-        const visibilityCodeToggler = screen.getAllByRole("button", { name: "Show" })[1];
-        expect(visibilityCodeToggler).not.toBeNull();
+        // heading without "BETA"
+        const title = screen.getByRole("heading", {
+          name: /SUSE Linux Enterprise High Availability Extension 16.0 x86_64/,
+          level: 4,
+        });
+        const extensionNode = title.parentElement;
 
-        // only the end of the code is displayed
-        screen.getByText(/\*+ad42/);
-        // not the full code
-        expect(screen.queryByText(registeredAddonInfoMock[0].registrationCode)).toBeNull();
+        // description is displayed
+        within(extensionNode).getByText(addonInfoMock[0].description);
 
-        // after pressing the "Show" button
-        await user.click(visibilityCodeToggler);
-        // the full code is visible
-        screen.getByText(registeredAddonInfoMock[0].registrationCode);
+        // registration input field is displayed
+        within(extensionNode).getByLabelText("Registration code");
+
+        // submit button is displayed
+        const addonRegButton = within(extensionNode).getByRole("button", { name: "Register" });
+      });
+
+      describe("and they are registered", () => {
+        beforeEach(() => {
+          registeredAddonInfoMock = [
+            {
+              id: "sle-ha",
+              version: "16.0",
+              registrationCode: "INTERNAL-USE-ONLY-1234-ad42",
+            },
+          ];
+        });
+
+        it("renders them with its registration code partially hidden", async () => {
+          const { user } = installerRender(<ProductRegistrationPage />, { withL10n: true });
+
+          // the second "Show" button, the first one belongs to the base product registration code
+          const visibilityCodeToggler = screen.getAllByRole("button", { name: "Show" })[1];
+          expect(visibilityCodeToggler).not.toBeNull();
+
+          // only the end of the code is displayed
+          screen.getByText(/\*+ad42/);
+          // not the full code
+          expect(screen.queryByText(registeredAddonInfoMock[0].registrationCode)).toBeNull();
+
+          // after pressing the "Show" button
+          await user.click(visibilityCodeToggler);
+          // the full code is visible
+          screen.getByText(registeredAddonInfoMock[0].registrationCode);
+        });
```

Comment:
NP: While updating this, I had the feeling that such a test actually belongs to the missing RegistrationExtensions.test.tsx